broadinstitute/ddp-study-server
# DSM export upgrades - DDP-7581

## Context
This upgrades the export capability to support both tsv and xlsx files, and support splitting multiselect questions into one column per option.  It also makes some performance improvements and bugfixes in the handling of JSON data.  It also contains fixes/improvements outlined in DDP-7581.  The functionality has been shown and approved by AT study staff.  

I have focused my testing on AT, RGP, and PanCan

These changes require the frontend PR https://github.com/broadinstitute/ddp-angular/pull/1489 to be visible.

TO TEST:
 1. ensure you have the frontend PR checked out too
 2. load the participant list for the AT study
 3. press the 'download' button
 4. leave all options on defaults
 5. hit download
 6. confirm the file is downloaded as a tsv
 7. open the file in a spreadsheet editor
 8. Search for 'MEDICAL_HISTORY.DETERMINED', and observe that the options have been split into one-column-per-option

## Checklist

- [*] I have labeled the type of changes involved using the `C-*` labels.
- [*] I have assessed potential risks and labeled using the `R-*` labels.
- [*] I have considered error handling and alerts, and added `L-*` labels as needed.
- [*] I have considered security and privacy, and added `I-*` labels as needed
- [*] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [*] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [*] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

## FUD Score

_Overall, how are you feeling about these changes?_

- [*] :sweat_smile: There might be some issues here

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [*] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [*] I have written zero automated tests but have poked around locally to verify proper functionality

## Release
